lib/engine/route.rb

```diff
@@ -331,7 +333,12 @@ def revenue(suppress_check_other: false, supress_route_token_check: false)
           check_overlap!
           check_connected!
 
-          @game.revenue_for(self, stops)
+          rev = @game.revenue_for(self, stops)
+          @route_error = nil
```

Comment:
Why do you need to set route_error to nil here?

Reply:
I was thinking that when the route is extended (by clicking in the UI), the `Route` object would be updated, instead of a new one being created, but I don't think that can actually be the case since `@revenue` is cached here. Removed this in 10758570b